tools/c7n_azure/c7n_azure/policy.py

```diff
@@ -165,10 +160,18 @@ def provision(self):
 
         FunctionAppUtilities().deploy_dedicated_function_app(params)
 
+    def get_logs(self, start, end):
+        """Retrieve logs for the policy"""
+        raise NotImplementedError("subclass responsibility")
+
+    def validate(self):
+        """Validate configuration settings for execution mode."""
+
+    def _publish_functions_package(self, queue_name=None):
```

Comment:
Can we call this function from provision()? 
And for the EventGrid we can override it to deploy queue & then call AzureFunctionMode::_publish_functions_package()

Reply:
It is called from provision. And that is actually the scenario I'm using.